docs/guides/python/blender-render.mdx

```diff
@@ -0,0 +1,905 @@
+
+```bash
+uv add bpy==4.2.0 --optional ml
+```
+
+We'll organize our project structure like so:
+
+```text
++--src/
+|  +--services/
+|      +-- main.py
++--nitric.yaml
++--docker/
+|  +-- python.dockerfile
++--.gitignore
++--.python-version
++--pyproject.toml
++--README.md
+```
+
+## Creating the resources
+
+We'll start by creating a file to define our Nitric resources. For this project we'll need an API, Batch Job, and two buckets, one for the `.blend` files and one for the resulting renders. The API will interface with the buckets, while the Batch Job will handle the long-running renders.
+
+```python title:src/resources.py
+from nitric.resources import api, job, bucket
+
+main_api = api("main")
+
+renderer_job = job("render-image")
+
+blend_bucket = bucket("blend_files")
+rendered_bucket = bucket("rendered_bucket")
+```
+
+## Routes for our API
+
+Now that we have defined resources, we can import our API and add some routes to access the buckets. Start by importing the resources and adding permissions to the resources.
+
+```python title:src/services/main.py
+import json
+import requests
+
+from nitric.application import Nitric
+from nitric.context import HttpContext
+from nitric.resources import BucketNotificationContext
+
+from src.resources import rendered_bucket, main_api, blend_bucket, renderer_job
+
+readable_rendered_bucket = rendered_bucket.allow("read")
+readable_writeable_blend_bucket = blend_bucket.allow("write", "read")
+submittable_renderer_job = renderer_job.allow("submit")
+
+Nitric.run()
+```
+
+We'll then write a route for getting a file from the rendered bucket. These will get a signed download url and redirect the user to this url for downloading the content.
+
+```python title:src/services/main.py
+# !collapse(1:12) collapsed
+import json
+import requests
+
+from nitric.application import Nitric
+from nitric.context import HttpContext
+from nitric.resources import BucketNotificationContext
+
+from src.resources import rendered_bucket, main_api, blend_bucket, renderer_job
+
+readable_rendered_bucket = rendered_bucket.allow("read")
+readable_writeable_blend_bucket = blend_bucket.allow("write", "read")
+submittable_renderer_job = renderer_job.allow("submit")
+
+@main_api.get("/render/:file")
+async def get_image(ctx: HttpContext):
+  file_name = ctx.req.params['file']
+
+  download_url = await readable_writeable_blend_bucket.file(file_name).download_url(3600)
+
+  ctx.res.headers["Location"] = download_url
+  ctx.res.status = 303
+
+  return ctx
+
+Nitric.run()
+```
+
+The final route will be for adding a `.blend` file for rendering as well as the render settings for the scene. This will add the contents of the request to a file in the blend bucket. Writing directly to a bucket is limited to 4 MB, to avoid this we use an upload URL to increase the limit to 5 GB.
```

Comment:
If we can update the request for uploading the blend file into a redirect then we might be able to remove this note. It's worth noting that the size limit for uploading directly to AWS lambda is 6MB